From ee9e72281961c4372aa342e8081ef0ac01e014ea Mon Sep 17 00:00:00 2001
From: Chen Jun <chenjun102@huawei.com>
Date: Tue, 29 Nov 2022 11:52:58 +0000
Subject: [PATCH 0816/1398] [Huawei] mm/sharepool: use delete_spg_node to
 replace some repetitive code

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I650K6

--------------------------------

delete_spg_node can be use to replace the code snippets as follows:
list_del(&node->proc_node);
spg->proc_num--;

Signed-off-by: Chen Jun <chenjun102@huawei.com>
---
 mm/share_pool.c | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/mm/share_pool.c b/mm/share_pool.c
index a0c6b2915b83..b76813dc8659 100644
--- a/mm/share_pool.c
+++ b/mm/share_pool.c
@@ -1691,8 +1691,7 @@ int mg_sp_group_del_task(int tgid, int spg_id)
 	down_write(&spg->rw_lock);
 	if (list_is_singular(&spg->procs))
 		is_alive = spg->is_alive = false;
-	spg->proc_num--;
-	list_del(&spg_node->proc_node);
+	delete_spg_node(spg, spg_node);
 	sp_group_drop(spg);
 	up_write(&spg->rw_lock);
 	if (!is_alive)
@@ -4300,8 +4299,7 @@ int sp_group_exit(void)
 		/* a dead group should NOT be reactive again */
 		if (spg_valid(spg) && list_is_singular(&spg->procs))
 			is_alive = spg->is_alive = false;
-		spg->proc_num--;
-		list_del(&spg_node->proc_node);
+		delete_spg_node(spg, spg_node);
 		up_write(&spg->rw_lock);
 
 		if (!is_alive)
-- 
2.17.1

